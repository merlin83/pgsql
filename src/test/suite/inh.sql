---------------------------------------------------------------------------
--
-- inh.sql-
--    checks inheritance 
--
--
-- Copyright (c) 1994, Regents of the University of California
--
-- $Id: inh.sql,v 1.1.1.1 1996-07-09 06:22:30 scrappy Exp $
--
---------------------------------------------------------------------------

create table person (name text, age int4, location point);
create table man () inherits(person);
create table emp (salary int4, manager char16) inherits(person);
create table student (gpa float8) inherits (person);
create table stud_emp (percent int4) inherits (emp, student);
create table female_stud_emp () inherits(stud_emp);

-- attr order: name, age, location
select * from person;
select * from man;
-- attr order: name, age, location, salary, manager
select * from emp;
-- attr order: name, age, location, gpa
select * from student;
-- attr order: name, age, location, salary, manager, gpa, percent
select * from stud_emp;
select * from female_stud_emp;

insert into person values ('andy', 14, '(1,1)');
insert into emp values ('betty', 20, '(2, 1)', 1000, 'mandy');
insert into student values ('cy', 45, '(3, 2)', 1.9);
insert into stud_emp values ('danny', 19, '(3.3, 4.55)', 400, 'mandy', 3.9);
insert into man values ('fred', 2, '(0, 0)');
insert into female_stud_emp values ('gina', 16, '(10, 10)', 500, 'mandy', 3.0);

-- andy
select * from person;

-- betty
select * from emp;

-- cy
select * from student;

-- danny
select * from stud_emp;

-- fred
select * from man;

-- gina
select * from female_stud_emp;

-- andy, betty, cy, danny, fred, gina
select * from person*;

-- betty, danny, gina
select * from emp*;

-- cy, danny, gina
select * from student*;

-- danny, gina
select * from stud_emp*;

drop table female_stud_emp;
drop table stud_emp;
drop table student;
drop table emp;
drop table man;
drop table person;
