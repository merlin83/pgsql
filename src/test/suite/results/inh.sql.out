QUERY: create table person (name text, age int4, location point);
QUERY: create table man () inherits(person);
QUERY: create table emp (salary int4, manager char16) inherits(person);
QUERY: create table student (gpa float8) inherits (person);
QUERY: create table stud_emp (percent int4) inherits (emp, student);
QUERY: create table female_stud_emp () inherits(stud_emp);
QUERY: select * from person;
name  age  location  
----- ---- --------- 
QUERY: select * from man;
name  age  location  
----- ---- --------- 
QUERY: select * from emp;
name  age  location  salary  manager  
----- ---- --------- ------- -------- 
QUERY: select * from student;
name  age  location  gpa  
----- ---- --------- ---- 
QUERY: select * from stud_emp;
name  age  location  salary  manager  gpa  percent  
----- ---- --------- ------- -------- ---- -------- 
QUERY: select * from female_stud_emp;
name  age  location  salary  manager  gpa  percent  
----- ---- --------- ------- -------- ---- -------- 
QUERY: insert into person values ('andy', 14, '(1,1)');
QUERY: insert into emp values ('betty', 20, '(2, 1)', 1000, 'mandy');
QUERY: insert into student values ('cy', 45, '(3, 2)', 1.9);
QUERY: insert into stud_emp values ('danny', 19, '(3.3, 4.55)', 400, 'mandy', 3.9);
QUERY: insert into man values ('fred', 2, '(0, 0)');
QUERY: insert into female_stud_emp values ('gina', 16, '(10, 10)', 500, 'mandy', 3.0);
QUERY: select * from person;
name  age  location  
----- ---- --------- 
andy  14   (1,1)     
QUERY: select * from emp;
name   age  location  salary  manager  
------ ---- --------- ------- -------- 
betty  20   (2,1)     1000    mandy    
QUERY: select * from student;
name  age  location  gpa  
----- ---- --------- ---- 
cy    45   (3,2)     1.9  
QUERY: select * from stud_emp;
name   age  location    salary  manager  gpa  percent  
------ ---- ----------- ------- -------- ---- -------- 
danny  19   (3.3,4.55)  400     mandy    3.9           
QUERY: select * from man;
name  age  location  
----- ---- --------- 
fred  2    (0,0)     
QUERY: select * from female_stud_emp;
name  age  location  salary  manager  gpa  percent  
----- ---- --------- ------- -------- ---- -------- 
gina  16   (10,10)   500     mandy    3             
QUERY: select * from person*;
name   age  location    
------ ---- ----------- 
andy   14   (1,1)       
fred   2    (0,0)       
betty  20   (2,1)       
cy     45   (3,2)       
danny  19   (3.3,4.55)  
gina   16   (10,10)     
QUERY: select * from emp*;
name   age  location    salary  manager  
------ ---- ----------- ------- -------- 
betty  20   (2,1)       1000    mandy    
danny  19   (3.3,4.55)  400     mandy    
gina   16   (10,10)     500     mandy    
QUERY: select * from student*;
name   age  location    gpa  
------ ---- ----------- ---- 
cy     45   (3,2)       1.9  
danny  19   (3.3,4.55)  3.9  
gina   16   (10,10)     3    
QUERY: select * from stud_emp*;
name   age  location    salary  manager  gpa  percent  
------ ---- ----------- ------- -------- ---- -------- 
danny  19   (3.3,4.55)  400     mandy    3.9           
gina   16   (10,10)     500     mandy    3             
QUERY: drop table female_stud_emp;
QUERY: drop table stud_emp;
QUERY: drop table student;
QUERY: drop table emp;
QUERY: drop table man;
QUERY: drop table person;
